### PR TITLE
`argparse` docs: normalize constant references

### PR DESCRIPTION
**Pull request by smontanaro**
Second attempt. In the future, please give me the opportunity to respond to comments before summarily closing a PR. It just seems rude to me.

### CODE REVIEW

File: Doc/library/argparse.rst [hunk 1]
```diff
@@ -1926,8 +1926,8 @@ FileType objects
       Namespace(out=<_io.TextIOWrapper name='file.txt' mode='w' encoding='UTF-8'>, raw=<_io.FileIO name='raw.dat' mode='wb'>)
 
    FileType objects understand the pseudo-argument ``'-'`` and automatically
-   convert this into ``sys.stdin`` for readable :class:`FileType` objects and
-   ``sys.stdout`` for writable :class:`FileType` objects::
+   convert this into :data:`sys.stdin` for readable :class:`FileType` objects and
+   :data:`sys.stdout`` for writable :class:`FileType` objects::
```
Comment (AlexWaygood) [inline Doc/library/argparse.rst]:
```suggestion
   :data:`sys.stdout` for writable :class:`FileType` objects::
```

Comment (smontanaro) [reply Doc/library/argparse.rst]: Thanks. I changed it in my repo, but can't push, even after following the hints in the git push message.

Comment (AlexWaygood) [reply Doc/library/argparse.rst]: I'll push it to your fork.

File: Doc/library/argparse.rst [hunk 2]
```diff
@@ -201,9 +201,9 @@ ArgumentParser objects
    * usage_ - The string describing the program usage (default: generated from
      arguments added to parser)
 
-   * description_ - Text to display before the argument help (default: none)
+   * description_ - Text to display before the argument help (default: ``None``)
 
-   * epilog_ - Text to display after the argument help (default: none)
+   * epilog_ - Text to display after the argument help (default: ``None``)
```
Comment (AlexWaygood) [inline Doc/library/argparse.rst]: Here, where the current docs say "default: none", I'm not sure the intent is actually to say "the default is the `None` constant". I think the intent is for it to read more like normal English prose, i.e.: "The `description` parameter determines the text to display before the argument help. The default is for no text to be displayed before the argument help."

File: Doc/library/argparse.rst (hunk 1 above)
Comment (AlexWaygood) [reply Doc/library/argparse.rst]: It might be that you made a change to the git branch using the GitHub website, and then forgot to `git pull` locally (to update your local branch with the changes you made remotely) before you `git push`ed? That's a common problem everybody runs into :)

**Comment by smontanaro** [comment]
I am happy to respond to comments, but i can no longer push changes. What is going on?

**Comment by AlexWaygood** [comment]
> I am happy to respond to comments, but i can no longer push changes. What is going on?

It might be that you made a change to the git branch using the GitHub website, and then forgot to git pull locally (to update your local branch with the changes you made remotely) before you git pushed? That's a common problem everybody runs into :)

**Comment by smontanaro** [comment]
Thanks @AlexWaygood. Here's what I'm seeing:

```
(python310) ~/src/python/cpython% git push origin HEAD
To github.com:smontanaro/cpython.git
 ! [rejected]              HEAD -> argparse-doc (non-fast-forward)
error: failed to push some refs to 'github.com:smontanaro/cpython.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
```

As far as I can tell from `git pull`, syncing `smontanaro/cpython` and `git fetch upstream`, I'm completely in sync, but I can't push. Part of my problem (at least by my way of thinking) is that it's telling me there's a problem, but doesn't give me a way to visualize it.

**Comment by AlexWaygood** [comment]
A bare `git pull` usually works in situations like this. You need to synchronise the "argparse-doc" branch of your local clone of `smontanaro/cpython` with the "argparse-doc" branch of `smontanaro/cpython` as it exists on GitHub

File: Doc/library/argparse.rst (hunk 2 above)
Comment (CAM-Gerlach) [reply Doc/library/argparse.rst]: If this is indeed the case, it could be made more clear and avoid confusion by saying something like `(default is no text)`

Comment (AlexWaygood) [reply Doc/library/argparse.rst]: > If this is indeed the case, it could be made more clear and avoid confusion by saying something like `(default is no text)`

Indeed, the current wording is slightly ambiguous; we don't disagree there

Comment (smontanaro) [reply Doc/library/argparse.rst]: Yes, thanks. That was the change I made on my local branch. I have still been unable to figure out why I can't push. I also don't know how to coax git into displaying some visual that will show me where my pulling, pushing, fetching, and syncing went wrong.

This seems low priority. Give me another day or two of head scratching before taking over. Thanks...

File: Doc/library/argparse.rst (hunk 1 above)
Comment (CAM-Gerlach) [reply Doc/library/argparse.rst]: > I changed it in my repo, but can't push, even after following the hints in the git push message.

If you haven't made other changes that aren't suggestions here, you can also just apply the suggestion(s) you want and commit directly here, which is typically the preferred approach. You go to the `Files` tab, click the `Add to batch` button on the suggestions you want, then click `Commit` with a suitable message.

> I'll push it to your fork.

The other solution would be @smontanaro could simply `git push -f` and then redo the merge with `main` if necessary, but he would have to cherry-pick that commit then. The problem now is that if he's also made the change (and possibly others) locally, there may be a merge conflict due to this, which makes things a lot harder to resolve.

> That's a common problem everybody runs into :)

Yup, I see this a lot as well and used to run into it in the past; I avoid it by getting in the habit of always `git pull`ing before making more changes to an existing PR branch (unless I'm absolutely certain there haven't been any) and always rebasing/merging locally

**Comment by CAM-Gerlach** [comment]
If you don't have any local changes you want to keep, there's always the shotgun approach, resetting your local branch to what's on the remote branch:

```shell
git fetch origin
git reset --hard origin/argparse-doc
```

Then you can commit, push, etc. as before. If you do have any local changes, you can either cherry-pick or rebase them on top.

**Comment by smontanaro** [comment]
I still have no idea how my local copy got weird w.r.t. the github version, but it's back in shape I think.

File: Doc/library/argparse.rst [hunk 3]
```diff
@@ -201,9 +201,10 @@ ArgumentParser objects
    * usage_ - The string describing the program usage (default: generated from
      arguments added to parser)
 
-   * description_ - Text to display before the argument help (default: none)
+   * description_ - Text to display before the argument help (default:
+     no text)
 
-   * epilog_ - Text to display after the argument help (default: none)
+   * epilog_ - Text to display after the argument help (default: no text)
```
Comment (CAM-Gerlach) [inline Doc/library/argparse.rst]:
```suggestion
   * epilog_ - Text to display after the argument help (by default, no text)
```

I might suggest this form to further avoid implying this as the default value to this named _argument_, as opposed to the _behavior_ by default

File: Doc/library/argparse.rst [hunk 4]
```diff
@@ -201,9 +201,10 @@ ArgumentParser objects
    * usage_ - The string describing the program usage (default: generated from
      arguments added to parser)
 
-   * description_ - Text to display before the argument help (default: none)
+   * description_ - Text to display before the argument help (default:
+     no text)
```
Comment (CAM-Gerlach) [inline Doc/library/argparse.rst]:
```suggestion
   * description_ - Text to display before the argument help
     (by default, no text)
```

Same here

**Review by CAM-Gerlach — COMMENTED** [review]
One small set of suggestions; other than that LGTM

**Comment by CAM-Gerlach** [comment]
It seems things have gone wrong here again with the commit history due to another local merge—all the commits are duplicated twice. In addition, this is on top of (as was the case originally), the original changes were implemented, manually reverted and redone for unclear reasons, and the rework to remove the const changes, plus the repeated sets of suggestion applications, all of which makes the commit history here very hard to follow.

To fix all of this for now, you can just reset your branch to the latest merge with master and apply just the delta in the working tree as a new commit. To do so (tested locally to confirm):

```python
git pull
git reset bded5edd9ab  # Last commit on main this has been merged with
git commit -am "Doc: Normalize constant references & link sys.stdin/stdout"
git push -f
```

Also, tip for the future: To avoid the time/effort, extra noise and CI resources of applying each suggestion individually, in the `Files` tab you can click `Add to batch` on each suggestion you want, then click `Commit` (adding an appropriately descriptive message) to commit them all in one go.

**Comment by smontanaro** [comment]
I followed your advice on the reset etc stuff. Does it look okay now?

I'm not sure what you're referring to when you say "files" tab. I do everything from the command line on my laptop, so always work locally. I'll take a look though and see if i can figure out what you mean.

**Comment by smontanaro** [comment]
Also, the GitHub app (at least on Android) seems to suck... Is it just me?

**Review by CAM-Gerlach — APPROVED** [review]
LGTM now @smontanaro , thanks!

**Comment by CAM-Gerlach** [comment]
> I followed your advice on the reset etc stuff. Does it look okay now?

Yup, looks great, thanks :+1: 

> I'm not sure what you're referring to when you say "files" tab. I do everything from the command line on my laptop, so always work locally. I'll take a look though and see if i can figure out what you mean.

I do as well as far as Git is concerned, and also basic GitHub stuff with the `hub` CLI (there's also the newer `gh` CLI), but what I'm referring to is PR review suggestions in the GitHub web interface. When you apply review suggestions, as it appears you did for the ones on [this review](https://github.com/python/cpython/pull/98765#pullrequestreview-1158866713), instead of clicking the `Commit` button on each suggestion, entering a message (or leaving the generic one), and clicking `Apply`, you can instead go to the [`Files Changed`](https://github.com/python/cpython/pull/98765/files) tab on the PR, click `Add to batch` on each suggestion you want to apply, and then when you've clicked the button on all of the ones you want, you click the `Commit` button at the top or on each one, enter a descriptive message and then click `Apply` to commit them all in one go. See [the docs](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request) for more details and screenshots.

Its no big deal if you do of course; I just wanted to mention the tip since a lot of devs aren't aware of that feature but it can save non-trivial time and effort especially with a larger number of suggestions.

> Also, the GitHub app (at least on Android) seems to suck... Is it just me?

I've never used any apps for GitHub myself (either mobile or desktop) other than the `hub` CLI, and I've never logged into the GitHub web UI on mobile (just very occasional browsing/checking things), so I couldn't say—I'm pretty old school myself. But I haven't heard the best things about it from some of the other devs who do.

**Comment by smontanaro** [comment]
>
> When you apply review suggestions, as it appears you did for the ones on this
> review
> <https://github.com/python/cpython/pull/98765#pullrequestreview-1158866713>,
> instead of clicking the Commit button on each suggestion, entering a
> message (or leaving the generic one), and clicking Apply, you can instead
> go to the Files Changed
> <https://github.com/python/cpython/pull/98765/files> tab on the PR, click Add
> to batch on each suggestion you want to apply, and then when you've
> clicked the button on all of the ones you want, you click the Commit
> button at the top or on each one, enter a descriptive message and then
> click Apply to commit them all in one go. See the docs
> <https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request>
> for more details and screenshots.Message ID:
> ***@***.***>
>

Ah, thanks. Ignore the email I just sent...

**Comment by CAM-Gerlach** [comment]
Sorry for all the trouble on this one!

**Review by AlexWaygood — APPROVED** [review]

**Comment by AlexWaygood** [comment]
Thanks, Skip!
